boards/arduino-leonardo/leonardo-runner.sh

```diff
@@ -1,6 +1,39 @@
 #!/usr/bin/env sh
 set -e
 
+LINUX_OS="LINUX"
+MAC_OS="MAC"
+UNKNOWN_OS="UNKNOWN"
+
+case "$(uname -s)" in
+    Linux*)     OS=$LINUX_OS;;
+    Darwin*)    OS=$MAC_OS;;
+    *)          OS=$UNKNOWN_OS
+esac
+
+if ! command -v numfmt &> /dev/null
+then
+    echo "required numfmt could not be found!"
+    echo "please install https://command-not-found.com/numfmt"
+    exit 1
+fi
+
+if ! command -v avrdude &> /dev/null
+then
+    echo "required avrdude could not be found!"
+    echo "please install https://command-not-found.com/avrdude"
+    exit 1
+fi
+
+if [ $OS = $LINUX_OS ]; then
+    ARDUINO_UPLOAD_PORT="/dev/ttyACM0"
+elif [ $OS = $MAC_OS ]; then
+    ARDUINO_UPLOAD_PORT="/dev/cu.usbmodem146201"
+else
+    echo "Not supported OS!"
+    exit 1    
+fi
```

Comment:
Let's call the variable `SERIAL_PORT`.  Also, should we really not allow "unsupported" OSes at all?  I'd rather try _something_ and if it fails, the user has to fix it.  e.g.

```sh
echo "unsupported OS, things might not work" >&2
SERIAL_PORT="/dev/ttyACM0"
```